File: WOT-iOS/WOT-iOS/core/requests/WOTWebRequest/Session/WOTClearSessionRequest.m
//
//  WOTClearSessionRequest.m
//  WOT-iOS
//
//  Created by Pavel Yeshchyk on 6/5/15.
//  Copyright (c) 2015 Pavel Yeshchyk. All rights reserved.
//

#import "WOTClearSessionRequest.h"
#import "UserSession.h"

@implementation WOTClearSessionRequest

- (void)executeWithArgs:(NSDictionary *)args {

    [super executeWithArgs:args];
    
    NSManagedObjectContext *context = [[WOTCoreDataProvider sharedInstance] mainManagedObjectContext];
    [context performBlock:^{
        
        [UserSession removeObjectsByPredicate:nil inManagedObjectContext:context];

        NSError *error = nil;
        [context save:&error];
        
        if (self.callback) {
            
            self.callback(nil, error);
        }
    }];
}

@end

File: WOT-iOS/WOTData/WOTData/WebServices/requests/Session/WOTClearSessionRequest.swift
//
//  WOTClearSessionRequest.swift
//  WOTData
//
//  Created by Pavel Yeshchyk on 3/5/20.
//  Copyright © 2020 Pavel Yeshchyk. All rights reserved.
//

import Foundation

@objc
public class WOTClearSessionRequest : WOTWEBRequest {
    override public var method: String { return "POST" }

    @objc
    public class func modelClassName() -> String {
        return ""
    }

    //- (void)start:(WOTRequestArguments * _Nonnull)args {
    //
    //    [super start:args];
    //
    //    id<WOTCoredataProviderProtocol> dataProvider = [WOTTankCoreDataProvider sharedInstance];
    //    NSManagedObjectContext *context = [dataProvider mainManagedObjectContext];
    //    [context performBlock:^{
    //
    //        [UserSession removeObjectsByPredicate:nil inManagedObjectContext:context];
    //
    //        NSError *error = nil;
    //        [context save:&error];
    //
    //        if (self.callback) {
    //
    //            self.callback(nil, error, nil);
    //        }
    //    }];
    //}
}
